app/controllers/api/v1/planning_applications_controller.rb

```diff
@@ -31,7 +31,7 @@ def create
         local_authority_id: @current_local_authority.id,
         boundary_geojson: (params[:boundary_geojson].to_json if params[:boundary_geojson].present?),
         proposal_details: (params[:proposal_details].to_json if params[:proposal_details].present?),
-        constraints: (params[:constraints].to_json if params[:constraints].present?),
+        constraints: (params[:constraints].present? ? params[:constraints].to_unsafe_hash.collect { |key, value| key if value }.compact : []),
```

Comment:
maybe this should be broken into a private method on the controller called something like `constraints_array_from_param` or something?

Reply:
I've done this now and yes, much better :)